### PR TITLE
Week3 Step 6

### PR DESCRIPTION
**Pull request by natalie-abreu**
[Link to portfolio](https://natalieabreu-step-2020.uc.r.appspot.com/index.html)
Added selection option to set limit on number of comments displayed
Added clear comment button which clears the entities in datastore

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -21,6 +21,8 @@
 import com.google.appengine.api.datastore.PreparedQuery;
 import com.google.appengine.api.datastore.Query;
 import com.google.appengine.api.datastore.Query.SortDirection;
+ import com.google.appengine.api.datastore.FetchOptions;
```
Comment (lazau-g) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:24]: nitpick: remove leading space. Remove newline.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 2]
```diff
@@ -35,18 +37,23 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+    static String dataType = "Comment";
```
Comment (kavinkarthik92) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:40]: Nit: Shouldn't this be aligned?

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 3]
```diff
@@ -76,4 +83,16 @@ private String convertToJsonUsingGson(List<String> messages) {
     String json = gson.toJson(messages);
     return json;
   }
+
+  private int getNumComments(HttpServletRequest request) {
+      String userInput = request.getParameter("max");
```
Comment (kavinkarthik92) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:88]: Move "max" to a static const. We might change the parameter name in future and this should not break the server logic.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 4]
```diff
@@ -76,4 +83,16 @@ private String convertToJsonUsingGson(List<String> messages) {
...
+      int numComments;
+      try {
+        numComments = Integer.parseInt(userInput);
+      } catch (NumberFormatException e) {
+        System.err.println("Could not convert to int: " + userInput);
+        return 5;
```
Comment (kavinkarthik92) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:94]: Why do we return 5 here?

Comment (natalie-abreu) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:94]: I just used 5 as a default value so it sets to if the user doesn't select anything

**Pull request merged**